### ISSUE TITLE
Processing commands during an Inventory#listener callback fails

### ISSUE THREAD

**Issue by codeHusky** [SpongePowered/SpongeForge#1679]
https://hastebin.com/kacokizogu.css

https://github.com/codeHusky/HuskyUI-Plugin/blob/master/src/main/java/com/codehusky/huskyui/states/action/CommandAction.java#L36

It's an unwinding phase incompatibility. Don't know enough about the cause tracker to explain it very well.

**Comment by phit** [comment]
@codeHusky can you add some more info please as to what API version this is for and update the hastebin link, as the error log is offline

**Comment by codeHusky** [comment]
well that.. sucks. I don't have that stored anywhere, but it was definitely the most recent SV at the time

**Comment by JoeSGT** [comment]
If i recall it was something like this https://pastebin.com/we7uhi6p

**Comment by codeHusky** [comment]
@JoeSGT Error log, not the code.

**Comment by JoeSGT** [comment]
oh  my bad sorry

**Comment by ryantheleach** [comment]
@codeHusky Sponge Vanilla or Sponge Forge? this is potentially looking indicative of a SpongeCommon bug.

**Comment by codeHusky** [comment]
I was using SpongeVanilla

**Comment by phit** [comment]
```java
@Plugin(id = "commandsthing ", name = "commands Test", description = "phit be testin")
public class commandsthing {
    @Listener
    public void onInventoryClick(ClickInventoryEvent event, @First Player player) {
        Transaction<ItemStackSnapshot> clickTransaction = event.getTransactions().get(0);
        ItemStack item = clickTransaction.getOriginal().createStack();
        if (item.getType().equals(ItemTypes.FIRE_CHARGE)) {
            //this will create an error in console without having a delayTicks
            Sponge.getCommandManager().process(player, "help");
        }
    }
}
```

works fine on 1.12.2-2529-7.0.0-BETA-2747, no errors either

**Comment by ryantheleach** [comment]
@phit Inventory::Listener is not listening to the on Inventory Click event.

It's an API we expose as a helper.

https://github.com/SpongePowered/SpongeAPI/blob/bleeding/src/main/java/org/spongepowered/api/item/inventory/Inventory.java#L629

**Comment by codeHusky** [comment]
Make sure these changes get into a 1.10.2 build.

**Comment by phit** [comment]
can you provide a log for this bug in 1.10 and add some code how to reproduce it, as I have no interest myself in doing any further testing for 1.10

the whole tracker system was [overhauled](https://github.com/SpongePowered/SpongeAPI/pull/1370) for 1.12 and that will never get a 1.10 build as there are breaking API changes, which I assume fixed this somewhere along the way

**Comment by codeHusky** [comment]
ill look into this in a bit. this was mostly an issue which affected HuskyUI, so I'm just trying to make sure it's resolved in all the versions I expect users to use HuskyUI with.

**Comment by phit** [comment]
Closing this issue, as of today 1.10.2 and 1.11.2 are not longer supported. If a sufficiently, decent, and tested Pull Request is made to fix this bug/issue, it will still be considered, however, it is NOT to be expected that any bug fixes made for 1.12.2 will be back ported to 1.11.2 or 1.10.2.

If you can reproduce this issue with SpongeForge on Minecraft 1.12.2 and this issue was wrongly closed, please [open a new issue](https://github.com/SpongePowered/SpongeForge/issues/new) with updated information.
